src/glob.zig

```diff
@@ -0,0 +1,2555 @@
+
+                    // `recursion_idx_bump` is the component idx to offset to start at
+                    // when handling this directory's contents
+                    const recursion_idx_bump: u32 = recursion_idx_bump: {
+                        // Handle double wildcard `**`, this could possibly
+                        // propagate the `**` to the directory's children
+                        if (pattern.syntax_hint == .Double) {
+                            // Stop the double wildcard if it matches the pattern afer it
+                            // Example: src/**/*.js
+                            // - Matches: src/bun.js/
+                            //            src/bun.js/foo/bar/baz.js
+                            if (!is_last and this.matchPatternImpl(next_pattern.?, entry_name)) {
+                                // But if the next pattern is the last
+                                // component, it should match and propagate the
+                                // double wildcard recursion to the directory's
+                                // children
+                                if (component_idx + 1 == this.patternComponents.items.len - 1) {
+                                    try this.appendMatchedPath(entry_name, dir_path, in_root);
+                                    break :recursion_idx_bump 0;
+                                }
+
+                                // In the normal case skip over the next pattern
+                                // since we matched it, example:
+                                // BEFORE: src/**/node_modules/**/*.js
+                                //              ^
+                                //  AFTER: src/**/node_modules/**/*.js
+                                //                             ^
+                                break :recursion_idx_bump 2;
+                            }
+
+                            if (is_last) {
+                                // Add this directory
+                                try this.appendMatchedPath(entry_name, dir_path, in_root);
+                            }
+
+                            break :recursion_idx_bump 0;
+                        }
+
+                        const matches = this.matchPatternImpl(&pattern, entry_name);
+                        if (matches) {
+                            if (is_last) {
+                                try this.appendMatchedPath(entry_name, dir_path, in_root);
+                                continue;
+                            }
+                            break :recursion_idx_bump 1;
+                        }
+                        continue;
+                    };
+
+                    const subdir_parts: []const []const u8 = &[_][]const u8{
+                        dir_path[0..dir_path.len],
+                        entry_name,
+                    };
+
+                    const subdir_entry_name = try std.fs.path.join(this.arena.allocator(), subdir_parts);
```

Comment:
Can we use bun's file path join logic instead of `std.fs.path`? I don't generally trust the zig standard library for string-parsing related code in it's current level of stability. It would be unsurprising if it didn't work with Windows' UTF-16 ill-formed surrogates or otherwise handle unicode correctly.